edgedb/datatypes/object.c

```diff
@@ -338,8 +338,44 @@ object_dir(EdgeObject *o, PyObject *args)
 }
 
 
+static PyObject *
+object_deepcopy(EdgeObject *x, PyObject *arg)
+{
+    Py_ssize_t size = Py_SIZE(x);
+    PyObject *y = PyDict_New();
+
+    Py_ssize_t i;
+    PyObject *name;
+    PyObject *item;
+    for (i = 0; i < size; i++) {
+        assert(x->ob_item[i] != NULL);
+        item = _py_copy_deepcopy(x->ob_item[i], arg);
+        if (item == NULL) {
+            Py_DECREF(y);
+            return NULL;
+        }
+        name = EdgeRecordDesc_PointerName(x->desc, i);
+        if (name == NULL) {
+            Py_DECREF(item);
+            Py_DECREF(y);
+            return NULL;
+        }
+        if (PyDict_SetItem(y, name, item)) {
+            Py_DECREF(name);
+            Py_DECREF(item);
+            Py_DECREF(y);
+            return NULL;
+        }
+        Py_DECREF(name);
+        Py_DECREF(item);
+    }
+    return y;
```

Comment:
I think it's somewhat a violation that we return vanilla Python types when our types get deepcopied. Why?